src/components/Provider/ProviderHeader/CopyTraderButton/CopyTraderButton.js

```diff
@@ -60,16 +102,17 @@ const CopyTraderButton = ({ provider }) => {
       flexDirection="row"
       justifyContent="flex-start"
     >
-      {provider.disable ? (
+      {disabled && !disconnecting && (
         <CustomButton className="submitButton" onClick={startCopying}>
           <FormattedMessage id="copyt.copythistrader" />
         </CustomButton>
-      ) : !followingFrom ||
-        provider.exchangeInternalId === storeSettings.selectedExchange.internalId ? (
+      )}
```

Comment:
In the futures I think it's more clear to do like that:
```
{disabled ? 
      <>
       {disconnecting && (...)}
       {!disconnecting && (!followingFrom || sameSelectedExchange) && (...)}
       </>
       : disconnecting && (...)
}
```
I haven't doubled checked the syntax but you get the idea

Reply:
I didn't do that because I was trying to avoid multiple nested conditions, this way I believe is more clear and easily understandable.